src/hooks/useGetRef.ts

```diff
@@ -0,0 +1,6 @@
+import { MutableRefObject, RefObject, useRef } from "react";
+
+export const useGetRef = <T extends HTMLElement>(maybeRef: RefObject<T> | undefined): MutableRefObject<T | null> => {
```

Comment:
Suggestion: Took me awhile to remember what this was for; maybe add an in-source comment like: `/** Used to sometimes accept a caller's ref, or if they did not pass one, use an internal one anyway. */`